include/net/net_l2.h

```diff
@@ -49,8 +49,9 @@ struct net_l2 {
 	 * This function is used by net core to push a packet to lower layer
 	 * (interface's L2), which in turn might work on the packet relevantly.
 	 * (adding proper header etc...)
+	 * Returns a negative error code, or the number of bytes sent otherwise.
```

Comment:
Driver patches as included in this commit don't seem to follow "the number of bytes sent otherwise", so stats would be off (unless I'm mixing up all the send()'s around).